docker-build-context/Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Goreleaser builds a docker image according to this Dockerfile by copying the sysl binary 
+# to a golang:1.13-alpine image and setting up the entrypoint.
+#
+# The produced image is published to https://hub.docker.com/r/anzbank/sysl
+ARG go_ver=1.14
+
+FROM golang:${go_ver}-buster as builder
+
+ARG sysl_ver=v0.7.1
+
+WORKDIR /work
+
+RUN git clone --branch ${sysl_ver} --single-branch --depth 1 https://github.com/anz-bank/sysl.git && \
```

Comment:
I think you are assuming this will be used in the context of a checked out repo mounted via a bind mount?

That would not allow auto builds to occur on Docker Hub. If it works differently for you let me know.

Reply:
Personal preference. I prefer to document the check out in the Dockerfile rather than relying on having the correct version lying around in the build context.

Will follow the product conventions.

Reply:
Well when I copy the source in via a COPY it does not build. The build error is

```
build github.com/anz-bank/sysl/cmd/sysl: cannot load github.com/anz-bank/sysl/pkg/cfg: no matching versions for query "latest"
```

(This is using a fresh clone at the v0.7.1 tag)

Cloning the same tag into the image and building works fine :-(.

Reply:
Yeah, the `.dockerignore` is broken. You'll need to remove `pkg` from it (and may as well remove the `!pkg/sysl_js` line too since that's now useless).

I added `pkg/antlr-4.7-complete.jar` and `scripts/generate` to it since they aren't needed for the docker build and just bloat the context.